larq/models.py

```diff
@@ -47,91 +67,316 @@ def _get_output_shape(layer):
+            raise NotImplementedError()
+
+    @property
+    def unique_param_bidtwidths(self):
+        return sorted(set([p.bitwidth for p in self.parameter_profiles]))
+
+    @property
+    def unique_op_precisions(self):
+        return sorted(set([op.precision for op in self.op_profiles]))
+
+    def generate_table_row(self, table_config):
+        row = [self._layer.name, self.input_precision(), self.output_shape]
+        for i in table_config["param_bidtwidths"]:
+            n = self.parameter_count(i)
+            n = _format_table_entry(n, table_config["param_units"])
+            row.append(n)
+        row.append(_format_table_entry(self.memory, table_config["memory_units"]))
+        for i in table_config["mac_precisions"]:
+            n = self.op_count("mac", i)
+            n = _format_table_entry(n, table_config["mac_units"])
+            row.append(n)
+
+        return row
+
+    def _quantized_weights(self):
+        try:
+            return self._layer.quantized_latent_weights
+        except:
+            return []
+
+    def _get_bitwidth(self, weight):
+        try:
+            quantizer = self._layer.quantizers[self._quantized_weights().index(weight)]
+            return quantizer.precision
+        except:
+            return 32
+
+
+class ModelProfile:
+    def __init__(self, model):
+        self.layer_profiles = [LayerProfile(l) for l in model.layers]
+
+    @property
+    def memory(self):
+        return sum(l.memory for l in self.layer_profiles)
+
+    @property
+    def fp_equivalent_memory(self):
+        return sum(l.fp_equivalent_memory for l in self.layer_profiles)
+
+    def parameter_count(self, bitwidth=None, trainable=None):
+        return sum(l.parameter_count(bitwidth, trainable) for l in self.layer_profiles)
+
+    def op_count(self, op_type=None, bitwidth=None):
+        return sum(l.op_count(op_type, bitwidth, 0) for l in self.layer_profiles)
+
+    @property
+    def unique_param_bidtwidths(self):
+        return sorted(
+            set(_flatten([l.unique_param_bidtwidths for l in self.layer_profiles]))
+        )
 
-def _generate_table(model):
-    layer_stats = [_parse_params(l) for l in model.layers]
-    summed_stat = _sum_params(layer_stats)
+    @property
+    def unique_op_precisions(self):
+        return sorted(
+            set(_flatten([l.unique_op_precisions for l in self.layer_profiles]))
+        )
 
-    table = [
-        [
+    def _generate_table_header(self, table_config):
+        return [
             "Layer",
             "Input prec.\n(bit)",
             "Outputs",
-            *(f"# {i}-bit" for i in sorted(summed_stat)),
-            "Memory\n(kB)",
+            *(
+                f"# {i}-bit\nx {table_config['param_units']}"
+                for i in table_config["param_bidtwidths"]
+            ),
+            f"Memory\n({_bitsize_as_str(table_config['memory_units'])})",
+            *(
+                f"{i}-bit macs\n({_bitsize_as_str(table_config['mac_units'])})"
```

Comment:
What do you think about
```suggestion
                f"{i}-bit MACs\n({_bitsize_as_str(table_config['mac_units'])})"
```